pages/slice/language-guide/class-types.md

```diff
@@ -178,6 +178,22 @@ When the generated code decodes a class in sliced format and slices off derived
 slices: it keeps them in the decoded instance. If you later send this class instance to another application that knows
 the derived class type, this application will decode successfully the full type.
 
+## Compact type IDs
+
+When the generated code encodes a class it needs a way to transmit the class's type.
+It does this by encoding the class's _type ID_ - a special value that is unique to each class.
+Normally, this is the class's fully-qualified identifier, which gets encoded as a string.
+However, to reduce overhead, Slice allows you to assign a numeric type ID to a class instead. For example:
+
+```slice
+class ClassWithStringTypeId {}     // has a type ID of "ClassWithStringTypeId"
+class ClassWithNumericTypeId(9) {} // has a type ID of `9`
+```
+
+Compact type IDs have no effect on the mapping of a class, or the contract for using it.
```

Comment:
> or the contract for using it.

I don't understand this "or".

For me, mapping of a class = API for the generated C# class. And this API = contract.

Reply:
I removed this distinction. It doesn't mention the "contract" any more.

But I do think there's a distinction:
For instance, `idempotent` only affects the contract, not the mapping.
Likewise, `[cs::identifier]` only affects the mapping, not the contract.

Reply:
I thought we used the term "contract" exclusively for b). Personally I wouldn't consider a) a 'contract'. That's just the API.

> When talking about Slice / IDL contract, we talk about (b). But here, you use contract for (a):

No, I was still talking about b), hence why I mentioned the mapping and contract separately.
There'd be no point to mentioning the mapping and the mapping contract separately.

Either way, this line was already simplified, so it doesn't mention the contract (talking about b)) at all anymore.